packages/apps-engine/src/server/AppManager.ts

```diff
@@ -492,6 +496,37 @@ export class AppManager {
         return true;
     }
 
+    public async migrate(id: string): Promise<boolean> {
+        const app = this.apps.get(id);
+
+        if (!app) {
+            throw new Error(`No App by the id "${id}" exists.`);
+        }
+
+        await app.call(AppMethod.ONUPDATE).catch((e) => console.warn('Error while migrating:', e));
+
+        await this.purgeAppConfig(app, { keepScheduledJobs: true });
+
+        const storageItem = await this.appMetadataStorage.retrieveOne(id);
+
+        app.getStorageItem().marketplaceInfo = storageItem.marketplaceInfo;
+        await app.validateLicense().catch();
```

Comment:
Calling `catch` and passing `undefined` still ends up triggering an unhandled rejection